hawk-spring/src/main/java/so/hawk/java/catcher/SpringHawk.java

```diff
@@ -0,0 +1,68 @@
+package so.hawk.java.catcher;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Manages uncaught exception handling for Spring Boot applications.
+ */
+public class SpringHawk extends CustomUncaughtExceptionHandler implements ApplicationContextAware {
```

Comment:
I don't think that main entry point should extend `CustomUncaughtExceptionHandler`